plugins/src/main/java/org/smartregister/fhir/gateway/plugins/Constants.java

```diff
@@ -15,9 +15,15 @@ public class Constants {
     public static final String PARAM_VALUES_SEPARATOR = ",";
     public static final String PROXY_TO_ENV = "PROXY_TO";
     public static final String TAG_SEARCH_PARAM = "_tag";
+    public static final String TYPE_SEARCH_PARAM = "type";
+    public static final String DEFAULT_ADMIN_LEVEL_TYPE_URL =
+            "https://smartregister.org/codes/administrative-level";
     public static final String AUTHORIZATION = "Authorization";
     public static final String KEYCLOAK_UUID = "keycloak-uuid";
     public static final String IDENTIFIER = "_id";
+    public static final String MIN_ADMIN_LEVEL = "administrativeLevelMin";
+    public static final String MAX_ADMIN_LEVEL = "administrativeLevelMax";
+    public static final int DEFAULT_MAX_ADMIN_LEVEL = 10;
```

Comment:
What is a reasonable default maximum admin level depth?

Reply:
I am not sure we can have a `max `level because countries have different hierarchies.  I know we can have the `min` which is `0` but I don't think we can restrict the max 

In case the `max` is not defined we are supposed to fetch the whole hierarchy to the end of the node

Reply:
Seeing this is how the type element is represented:
```
"type": {
    "coding": [
      {
        "system": "http://example.org/fhir/CodeSystem/location-type",
        "code": "1",
        "display": "Level 1"
      }
    ],
```
querying directly with conditions like `code is greater than 5` on a string representation of a code is not possible. FHIR does not support such numeric comparisons on string fields directly within the query. So, to handle this, I have implement post-query filtering in the application logic.

This commit addresses that https://github.com/onaio/fhir-gateway-extension/pull/66/commits/7d9ba8b5154cccb106491d475947e8427ffe4148 
This means, api requests with `adminstrativeLevelMax` set will be more optimized than those without.

Reply:
This makes sense. Can we review the performance with what we have as we think of how to address this?

Reply:
How many levels can we have at most? 

How would the performance compare if for example we used an OR operation for the levels since they are numeric? e.g. if we know we have 10 levels, the query for greater than 5 can be where code equals 6 or code equals 7 or code equals 8 e.t.c The assumption is that OR operation works this way.

Reply:
@ndegwamartin That was indeed the original implementation, based on the assumption that a maximum depth could be defined. The code constructed an `OR` query for all locations within the specified range, where the maximum value was determined as the greater of the `DEFAULT_MAX_ADMIN_LEVEL` constant and the `administrativeLevelMax` parameter provided by the user, and the minimum value was set to either `0` or the `administrativeLevelMin` parameter.

This functionality will continue to apply in all cases except when a maximum depth cannot be determined. Specifically, this occurs when neither `administrativeLevelMax` nor the `DEFAULT_MAX_ADMIN_LEVEL` constant are defined. As mentioned by @dubdabasoduba, this is a possibility and therefore, in such instances, post-processing will be necessary to filter the data appropriately.

Reply:
If the performance impact is significant when its open ended I think we are better off making a best guess estimate for a default max. I also think in real life hierarchies cannot go above 10 levels. We could even increase it to 15 or something. 

We can also use data from OpenSRP v1, current OpenSRP v2 and other systems e.g. OpenMRS known support/deployments for what max levels we have seen so far. 

On top of this, we could make that default configurable at runtime incase a deployment needs to change it cc 
 @lincmba @dubdabasoduba

Reply:
@ndegwamartin let's have the default at `DEFAULT_MAX_ADMIN_LEVEL=10` 

Also let's add this to the documentation.